ddl/delete_range.go

```diff
@@ -238,7 +238,23 @@ func insertJobIntoDeleteRangeTable(ctx sessionctx.Context, job *model.Job) error
 		}
 	case model.ActionDropTable, model.ActionTruncateTable:
 		tableID := job.TableID
-		startKey := tablecodec.EncodeTablePrefix(tableID)
+		// The startKey here is for compatibility with previous versions and old version did not handle endKey.
+		var startKey kv.Key
+		var partitionIDs []int64
+		if err := job.DecodeArgs(startKey, &partitionIDs); err != nil {
+			return errors.Trace(err)
+		}
+		if len(partitionIDs) > 0 {
+			for _, pid := range partitionIDs {
+				startKey = tablecodec.EncodeTablePrefix(pid)
+				endKey := tablecodec.EncodeTablePrefix(pid + 1)
+				if err := doInsert(s, job.ID, pid, startKey, endKey, now); err != nil {
+					return errors.Trace(err)
+				}
+			}
+			return nil
+		}
+		startKey = tablecodec.EncodeTablePrefix(tableID)
```

Comment:
if the table has partitions, we still need to add del range that encodes from tableID?

Reply:
@winkyao No, it will return on line 255.